StudentManagement5/src/test/java/raisetech5/StudentManagement5/repository/StudentRepositoryTest.java

```diff
@@ -1,23 +1,199 @@
 import raisetech5.StudentManagement5.data.Student;
+import raisetech5.StudentManagement5.data.StudentCourse;
+import raisetech5.StudentManagement5.domain.StudentDetail;
+import raisetech5.StudentManagement5.service.StudentService;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
 
 @MybatisTest
 class StudentRepositoryTest {
 
   @Autowired
   private StudentRepository sut;
+  @Autowired
+  private StudentRepository repository;
 
   @Test
   void 受講生の全件検索が行えること() {
     List<Student> actual = sut.search();
-    assertThat(actual.size()).isEqualTo(5);//20:00
+    assertThat(actual.size()).isEqualTo(0);
   }
-}
\ No newline at end of file
+
+  @Test
+  void 受講生の登録が行えること() {
+    Student student = new Student();
+    student.setName("榎並浩二");
+    student.setFurigana("エナミコウジ");
+    student.setNickname("エナミ");
+    student.setEmail("test@example.com");
+    student.setTown("奈良県");
+    student.setGender("男性");
+    student.setAge(38);
+    student.setRemark("");
+    student.setDeleted(false);
+
+    sut.registerStudent(student);
+
+    List<Student> actual = sut.search();
+
+    assertThat(actual.size()).isEqualTo(1);
+  }
+
+  @Test
+  void 受講生が検索できること() {
+
+    Student student = new Student();
+    student.setName("榎並浩二");
+    student.setFurigana("エナミコウジ");
+    student.setNickname("エナミ");
+    student.setEmail("test@example.com");
+    student.setTown("奈良県");
+    student.setGender("男性");
+    student.setAge(38);
+    student.setRemark("");
+    student.setDeleted(false);
+
+    sut.registerStudent(student);
+    String studentId = student.getId();
+
+    Student actual = sut.searchStudent(studentId);
+
+    assertThat(actual).isNotNull();
+    assertThat(actual.getName()).isEqualTo("榎並浩二");
+    assertThat(actual.getFurigana()).isEqualTo("エナミコウジ");
+    assertThat(actual.getEmail()).isEqualTo("test@example.com");
+    assertThat(actual.getTown()).isEqualTo("奈良県");
+    assertThat(actual.getGender()).isEqualTo("男性");
+    assertThat(actual.getAge()).isEqualTo(38);
+  }
+
+  @Test
+  void 受講生コースの全件検索ができること() {
+
+    List<StudentCourse> actual = sut.searchStudentCourseList();
+
+    assertThat(actual.size()).isEqualTo(6);
+
+    assertThat(actual).extracting(StudentCourse::getCoursName)
+        .containsExactlyInAnyOrder("Math", "History", "Science", "コース", "Javaコース",
+            "Javaコース");
+  }
+
+  @Test
+  void 受講生コース情報が検索できること() {
+
+    Student student = new Student();
+    student.setName("MisakiTakahashi");
+    student.setFurigana("Misaki Takahashi");
+    student.setNickname("Misaki");
+    student.setEmail("misaki@example.com");
+    student.setTown("Fukuoka");
+    student.setGender("Female");
+    student.setAge(21);
+    student.setRemark("");
+    student.setDeleted(false);
+
+    sut.registerStudent(student);
+    String studentId = student.getId();
+
+    StudentCourse studentCourse = new StudentCourse();
+    studentCourse.setStudent_id(studentId);
```

Comment:
キャメルケースに沿うと以下のようなメソッド名になりますね
修正お願いします。
```suggestion
    studentCourse.setStudentId(studentId);
```